### PR TITLE
Port Link tests from enzyme to RTL

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

Porting the `wonder-blocks-link` tests from `enzyme` to `@testing-library/react`.

Issue: XXX-XXXX

## Test plan:

Verify that all the unit tests pass.

### CODE REVIEW

File: packages/wonder-blocks-link/src/components/__tests__/link.test.js [hunk 1]
```diff
@@ -41,16 +35,16 @@ describe("Link", () => {
             );
 
             // Act
-            const linkWrapper = wrapper.find("Link").first();
-            linkWrapper.simulate("click", {button: 0});
+            const linkWrapper = screen.getByText("Click me!");
```
Comment (nishasy) [inline packages/wonder-blocks-link/src/components/__tests__/link.test.js:38]: Nit: I think this can be just `link` instead of `linkWrapper` now.

**Review by nishasy — APPROVED** [review]
Nice! Thanks for doing this! No blocking requests

File: packages/wonder-blocks-link/src/components/__tests__/link.test.js (hunk 1 above)
Comment (jandrade) [reply packages/wonder-blocks-link/src/components/__tests__/link.test.js:39]: ah yeah, good call! will update after fixing the failed test.

File: packages/wonder-blocks-link/src/components/__tests__/link.test.js [hunk 2]
```diff
@@ -179,21 +164,19 @@ describe("Link", () => {
...
 
             // Act
-            const linkWrapper = wrapper.find("Link").first();
-            linkWrapper.simulate("click", {
-                button: 0,
-            });
-            await wait(0);
-            wrapper.update();
+            const link = screen.getByText("Click me!");
+            userEvent.click(link);
```
Comment (kevinbarabash) [inline packages/wonder-blocks-link/src/components/__tests__/link.test.js:168]: So much easier. 🙂

File: packages/wonder-blocks-link/src/components/__tests__/link.test.js [hunk 3]
```diff
@@ -325,38 +298,38 @@ describe("Link", () => {
...
+                <Link href="/" onKeyUp={(e) => (keyCode = e.keyCode)}>
                     Click me!
                 </Link>,
             );
 
             // Act
-            wrapper.find(Link).simulate("keydown", {keyCode: 32});
+            const link = screen.getByText("Click me!");
+            // eslint-disable-next-line testing-library/prefer-user-event
+            fireEvent.keyUp(link, {keyCode: 32});
```
Comment (kevinbarabash) [inline packages/wonder-blocks-link/src/components/__tests__/link.test.js:329]: **question:** why did this change to a keyup from a keydown?

**Review by kevinbarabash — COMMENTED** [review]
Thanks for porting these tests to RTL. 💚

File: packages/wonder-blocks-link/src/components/__tests__/link.test.js (hunk 3 above)
Comment (jandrade) [reply packages/wonder-blocks-link/src/components/__tests__/link.test.js:329]: I think the previous test was giving false positives as we are intending to test `keyUp` and the previous version was testing `keyDown` instead.

**Pull request merged**